### PR TITLE
merge traffic splitting/shifting docs

### LINKED ISSUE #379: docs: consolidate traffic shifting docs
@rshriram not urgent but just opening this issue so we don't forget.

### PR DESCRIPTION
**Pull request by rshriram**
fixes #379 
cc @junr03

### CODE REVIEW

File: docs/configuration/http_conn_man/route_config/route.rst [hunk 1]
```diff
@@ -122,7 +122,9 @@ Runtime
 -------
 
 A :ref:`runtime <arch_overview_runtime>` route configuration can be used to roll out route changes
-in a gradual manner without full code/config deploys.
+in a gradual manner without full code/config deploys. Refer to
+:ref:`traffic shifting <config_http_conn_man_route_table_traffic_splitting_shift>` docs for examples
```
Comment (junr03) [inline docs/configuration/http_conn_man/route_config/route.rst]: docs ~for examples~ for additional documentation

File: docs/configuration/http_conn_man/route_config/traffic_splitting.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
+.. _config_http_conn_man_route_table_traffic_splitting:
+
+Traffic Shifting/Splitting
+===========================================
+
+Envoy's router can split traffic to a route in a virtual host across
+two or more upstream clusters. There are two common use cases. The first use
```
Comment (junr03) [inline docs/configuration/http_conn_man/route_config/traffic_splitting.rst]: I would do

There are two common use cases:

1.  Version Upgrades: where traffic to a route is...
2.  A/B testing or multivariate testing: where `two or more versions`...

File: docs/intro/arch_overview/http_routing.rst [hunk 3]
```diff
@@ -27,8 +27,10 @@ request. The router filter supports the following features:
 * Request timeout specified either via :ref:`HTTP
   header <config_http_filters_router_headers>` or via :ref:`route configuration
   <config_http_conn_man_route_table_route_timeout>`.
-* Runtime configuration routing rules. Runtime configuration can be used to create :ref:`traffic
-  shifting <config_http_conn_man_route_table_traffic_shifting>`.
+* Probability-based route selection via :ref:`runtimes <config_http_conn_man_route_table_route_runtime>`
```
Comment (junr03) [inline docs/intro/arch_overview/http_routing.rst]: runtimes -> runtime values?

Comment (rshriram) [inline docs/intro/arch_overview/http_routing.rst]: all done. Thanks for the feedback!

**Comment by junr03** [comment]
Looks great, thanks for merging these together.

**Pull request merged**